ui/Rhino/AGS/dev/AGS_session_load_cmd.py

```diff
@@ -51,15 +55,32 @@ def RunCommand(is_interactive):
 
         if 'form' in data and data['form']:
             form = FormDiagram.from_data(data['form'])
-            scene.add(form, name='Form', layer='AGS::FormDiagram')
+            form_id = scene.add(form, name='Form', layer='AGS::FormDiagram')
+            form = scene.find(form_id)
+
+            # calculate the scale factor for FormDiagram
+            drawingscale_forces = calculate_drawingscale_forces(form.diagram)
+            form.artist.settings['scale.forces'] = drawingscale_forces
```

Comment:
What if we  first search from the `drawing_scale_forces` and `scale_factor` already stored in the session? They should be stored within `form.artist.settings['scale.forces']` and `force.artist.scale` already. Also, when we load the force, there should already have some information about the position of the `anchor_point`. I think it is better try recovering these information before and then just ask for user selection/scale calculation if we don't find it...

Reply:
true. but i don't think any of that information is currently stored in the session. and the rhino-objects and the artist have no serialisation mechanism yet. working on that though... let's leave it like this for now and update as soon as the serialisation mechanisms are in place...

Reply:
Same as Tom said... these attributes are saved in the artist not in the diagram.... that's why they need to be reculated and the anchor needs to be chosen again.

Reply:
Ok!